### PR TITLE
Merge searchingUserCases with Master

### PR DESCRIPTION
**Pull request by sehoven**
US 04.01.01 to US 04.02.01 have tests. ( This closes issue [#5](https://github.com/CMPUT301F16T01/Carrier/issues/5) )

### CODE REVIEW

File: app/src/test/java/comcmput301f16t01/github/carrier/RequestsTests.java [hunk 1]
```diff
@@ -47,15 +47,15 @@ public void riderRequest() {
     @Test
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/RequestsTests.java:47]: Thanks for adding description to everything 👍

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        rc.addRequest(request1);
+        rc.addRequest(request2);
+        rc.addRequest(request3);
+
+        Location driverLocation = new Location(latitude4, longitude4);
+        // this method should return a list of requests, sorted based on proximity of start location
+        // for now I'm assuming there are limits on how far away a request can be to be included in this list
+        // TODO would it be better to use ArrayList<Request> or requestList
+        requests = rc.searchByLocation(driverLocation);
+        assertTrue("Search did not return 2 requests", requests.size() == 2);
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: Hold up. So the driverLocation is in Edmonton but it should return only the close-ish ones so why should requests return 2 requests, there's only one rider in Edmonton (Mandy). The others (Abigail and Alison) are way too far. Unless Calgary is close enough in this test then I guess that's fine.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        // TODO clarify our terminology...should this return open or accepted requests
+        Driver driver = new Driver("Amber");
+        rc.addDriver(request1, driver);
+        rc.acceptDriver(request1, driver);
+
+        // request1 should no longer be included in the search results
+        requests = rc.searchByLocation(driverLocation);
+        assertTrue("Search did not return 1 request", requests.size() == 1);
+        // check that the requests are ordered properly
+        assertEquals("Closest request incorrect", request3, requests.get(0));
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:80]: Same thing here. When we end up actually defining our proximity, I think Calgary might be too far.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        assertFalse("Search returned non-open request", requests.contains(request1));
+        assertFalse("Search returned location out of range", requests.contains(request2));
+    }
+
+    /**
+     * As a driver, I want to browse and search for open requests by keyword.
+     * Related: US 04.02.01
+     */
+    @Test
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:89]: I'm Sarah, I write long tests. Huuuuhhhhhh.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        Location endLocation3 = new Location(latitude3, longitude3);
+        String description3 = "Going home from school";
+        Request request3 = new Request(rider3, startLocation3, endLocation3, description3);
+
+        RequestController rc = new RequestController();
+        rc.addRequest(request1);
+        rc.addRequest(request2);
+        rc.addRequest(request3);
+
+        // this method should return a list of requests based on keywords in the request description
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:116]: Doesn't really have to do with your code but we should talk about what a keyword is.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        String query1 = "home";
+        String query2 = "whyte"; // should not be case-dependent
+        String query3 = "downtown";
+
+        // TODO should we allow the capability to search more than one keyword?
+        requests = rc.searchByKeyword(query1);
```
Comment (kieter) [inline app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:122]: 🌮

**Review by kieter — CHANGES_REQUESTED** [review]
I really only had one problem in SearchingTests.java but other than that just javadoc stuff and I'll accept the pull request 😎.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java (hunk 2 above)
Comment (sehoven) [reply app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: I was assuming for this test that Calgary was close enough to be included in the test. This is something we will eventually have to define more precisely.

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java (hunk 3 above)
Comment (sehoven) [reply app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:80]: This is fair. I just made a decision for writing the test. We should determine the exact range that we consider to be "close enough".

File: app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java (hunk 5 above)
Comment (sehoven) [reply app/src/test/java/comcmput301f16t01/github/carrier/SearchingTests.java:116]: I was going off of the description [here](https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=709813), but we may want to define this more precisely within our application. Another question may be if a keyword can be more than one word.

**Comment by sehoven** [comment]
@kieter Do you want me to change the range of the location field to be included in the search results (i.e. say that Calgary is too far) or is this good enough for the test?

**Comment by kieter** [comment]
I think it's good enough for the test. I also think it would be cool if the user can set the range of how far is too far. Like on Tinder.

**Comment by kieter** [comment]
Cool. Merging.

**Pull request merged**